client/client_test.go

```diff
@@ -118,17 +120,18 @@ func TestRegisterAndAddProbe(t *testing.T) {
 
 // TestRemoveProbe tests that the RemoveProbe() method removes a probe from Cloudprober
 func TestRemoveProbe(t *testing.T) {
+	ctx := context.Background()
 	client := setupCloudproberAndClient(t)
 	defer teardownCloudproberAndClient(t, client)
 
 	for i := 0; i < testRunCount; i++ {
 		probeName := fmt.Sprintf("testExtension%d", i)
-		if err := client.RegisterAndAddProbe(context.Background(), int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
+		if err := client.RegisterAndAddProbe(ctx, int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
 			t.Errorf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
 		}
 
 		// Check if the probe was added correctly
-		probesList, err := client.ListProbes(context.Background())
+		probesList, err := client.ListProbes(ctx)
 		if err != nil {
 			t.Errorf("ListProbes() failed, expected error: %v, got: %v", nil, err)
```

Comment:
This should be `Fatalf`.

Reply:
I disagree, `ListProbes()` is one of the methods under test here, and it is useful to see the results of the rest of the test rather than exiting here. For example, if `ListProbes()` fails in one specific test multiple times, but not in the others, that would be useful to know.